wire-ios-data-model/Source/Utilis/EAR/EARService.swift

```diff
@@ -139,6 +142,31 @@ public class EARService: EARServiceInterface {
         secondaryPublicKeyDescription = .secondaryKeyDescription(accountID: accountID)
         secondaryPrivateKeyDescription = .secondaryKeyDescription(accountID: accountID)
         databaseKeyDescription = .keyDescription(accountID: accountID)
+
+        if canPerformKeyMigration {
+            migrateKeysIfNeeded()
+        }
+    }
+
+    // MARK: - Migrate keys
+
+    private func migrateKeysIfNeeded() {
+        WireLogger.ear.info("migrating ear keys if needed...")
+
+        guard
+            databaseContexts.contains(where: \.encryptMessagesAtRest),
+            !existSecondaryKeys
+        else {
+            return
+        }
+
+        do {
+            try deleteSecondaryKeys()
```

Comment:
why do we try to delete if the secondaryKeys don't exist?

Reply:
It's just a precaution, or I guess a habit. I'll remove it.